Mathlib/Topology/Compactness/CompactSystem.lean

```diff
@@ -0,0 +1,174 @@
+/-
+Copyright (c) 2025 Peter Pfaffelhuber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne, Peter Pfaffelhuber
+-/
+import Mathlib.Data.Set.Dissipate
+import Mathlib.Topology.Separation.Hausdorff
+
+/-!
+# Compact systems.
+
+This files defines compact systems of sets.
+
+## Main definitions
+
+* `IsCompactSystem`: A set of sets is a compact system if, whenever a countable subfamily has empty
+  intersection, then finitely many of them already have empty intersection.
+
+## Main results
+
+* `IsClosedCompact.isCompactSystem`: The set of closed and compact sets is a compact system.
+* `IsClosedCompact.isCompactSystem_of_T2Space`: In a `T2Space α`, the set of compact sets
+  is a compact system in a `T2Space`.
+-/
+
+open Set Finset Nat
+
+variable {α : Type*} {p : Set α → Prop} {C : ℕ → Set α}
+
+section definition
+
+/-- A set of sets is a compact system if, whenever a countable subfamily has empty intersection,
+then finitely many of them already have empty intersection. -/
+def IsCompactSystem (p : Set α → Prop) : Prop :=
+  ∀ C : ℕ → Set α, (∀ i, p (C i)) → ⋂ i, C i = ∅ → ∃ (n : ℕ), Dissipate C n = ∅
+
+end definition
+
+namespace IsCompactSystem
+
+/-- In a compact system, given a countable family with empty intersection, we choose a finite
+subfamily with empty intersection. -/
+noncomputable
+def max_of_empty (hp : IsCompactSystem p) (hC : ∀ i, p (C i))
```

Comment:
I find this name strange (there is no `max` around). Also, would it make sense to use `Nat.find` in the definition, so that the witness is the smallest possible one?